kernel/http_client/src/lib.rs

```
@@ -89,237 +81,261 @@ impl HttpResponse {
-                    // which is necessary to attempt to parse it as an HTTP response.
-                    // Later, we can remove bytes towards the end if we ended up appending too many bytes,
-                    // e.g., we received more than enough bytes and some of them were for the next packet.
-                    packet_byte_buffer.extend_from_slice(data);
-
-                    let bytes_popped_off = {
-                        // Check to see if we've received the full HTTP response:
-                        // First, by checking whether we have received all of the headers (and can properly parse them)
-                        // Second, by getting the content length header and seeing if we've received the full content (in num bytes)
-                        let mut headers = [httparse::EMPTY_HEADER; 64];
-                        let mut response = httparse::Response::new(&mut headers);
-                        match response.parse(&packet_byte_buffer) {
-                            Ok(httparse::Status::Partial) => {
-                                trace!("http_client: received partial HTTP response...");
-                                // we haven't received all of the HTTP header bytes yet, 
-                                // so pop off all of the bytes from the recv buffer into our packet
-                                data.len()
-                            }
+            let locked = socket.lock();
+            let can_send = locked.can_send();
+            let can_recv = locked.can_recv();
+            let may_recv = locked.may_recv();
+            drop(locked);
+
+            state = match state {
+                HttpState::Requesting if can_send => {
+                    debug!("http_client: sending HTTP request: {:?}", request);
+                    socket
+                        .lock()
+                        .send_slice(request.as_ref())
+                        .expect("http_client: cannot send request");
```

is there a reason why we choose to panic in this particular case rather than return an error? IOW, is this is an unexpected error here that should never happen?

---

No, just a remnant of me copy-pasting the previous `send` implementation.